nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetHTTP.java

```diff
@@ -467,6 +486,18 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
             if (accept != null) {
                 get.addHeader(HEADER_ACCEPT, accept);
             }
+
+            // Add dynamic headers
+
+            PropertyValue customHeaderValue;
+            for (PropertyDescriptor customProperty : customHeaders) {
+                customHeaderValue = context.getProperty(customProperty).evaluateAttributeExpressions();
+                if (StringUtils.isNotBlank(customHeaderValue.getValue())) {
```

Comment:
I was working under two assumptions:

* onScheduled is called once per scheduling;
* in theory use time related expression language even if incoming flowfiles were to be absent

Does it make sense?